File: 153 Find Minimum in Rotated Sorted Array/find_min_in_rot_sorted_array.c
int findMin(int* nums, int numsSize) {
    int lo = 0, hi = numsSize - 1;
    while (lo < hi) {
        int mid = lo + (hi - lo) / 2;
        if (nums[lo] > nums[mid]) hi = mid;
        else if (nums[hi] < nums[mid]) lo = mid + 1;
        else break;
    }
    return nums[lo];
} // Author: Xinyu Chen

File: 153 Find Minimum in Rotated Sorted Array/find_min_in_rot_sorted_array.cs
public class Solution {
    public int FindMin(int[] nums) {
        int lo = 0, hi = nums.Length - 1;
        while (lo < hi) {
            int mid = lo + (hi - lo) / 2;
            if (nums[mid] < nums[lo]) hi = mid;
            else if (nums[mid] > nums[hi]) lo = mid + 1;
            else break;
        }
        return nums[lo];
    }
} // Author: XC